README.md

```diff
@@ -170,6 +170,33 @@ let peripherals = embassy_nrf::init(config);
 
 ## Troubleshooting
 
+If you are sure you have set interrupts correctly, but are still getting an error like below:
+```
+[ERROR]Location<lib.rs:104>panicked at 'sd_softdevice_enable err SdmIncorrectInterruptConfiguration'
+```
+
+You can use this code to print whether an interrupt is enabled, and its priority:
+```rust
+let interrupt_numbers = [
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
+    10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
+    20, 21, 22, 23, 24, 25, 26, 27, 28, 29,
+    32, 33, 34, 35, 36, 37, 38, 39, 40, 41,
+    42, 45, 47
+    ];
+
+for &num in &interrupt_numbers {
+    let interrupt = unsafe { mem::transmute::<u16, Interrupt>(num) };
+    let is_enabled = NVIC::is_enabled(interrupt);
+    let priority = NVIC::get_priority(interrupt);
+
+    println!("Interrupt {}: Enabled = {}, Priority = {}", num, is_enabled, priority.to_be_bytes());
```

Comment:
I think if you use `InterruptExt` you can get the nicer `Priority::Px` enum which doesn't require the user to shift the priority bits.

```rust
println!("Interrupt {}: Enabled = {}, Priority = {:?}", num, interrupt.is_enabled(), interrupt.get_priority());

Reply:
I'm a little lost here, could you please clarify for me? Thank you!

Reply:
if you use [this](https://docs.embassy.dev/embassy-nrf/git/nrf52840/interrupt/trait.InterruptExt.html#method.get_priority) `get_priority()` method instead of the one from `cortex-m`, you get an instance of [this](https://docs.embassy.dev/embassy-nrf/git/nrf52840/interrupt/enum.Priority.html) enum as a result, which is the nice 0-7 priority number, not a raw u8 with the priority encoded in the top 3 bits.

Reply:
So I think you meant do this?

```
for &num in &interrupt_numbers {
        let interrupt = unsafe { mem::transmute::<u16, Interrupt>(num) };
        let is_enabled = InterruptExt::is_enabled(interrupt);
        let priority = InterruptExt::get_priority(interrupt);

        println!("Interrupt {}: Enabled = {}, Priority = {:?}", num, is_enabled, priority);
    }
```

Only problem I get now is:
```
The trait bound `embassy_nrf::interrupt::Priority: defmt::Format` is not satisfied
```

Reply:
should be fixed by https://github.com/embassy-rs/embassy/pull/1787
if you enable the `defmt` feature in `embassy-nrf`.

Reply:
also `0..47` iterates until 46 only, you want `0..48`.

Reply:
Ok! I think we should be good to go now.

Thank you for your guidance!